src/util/range.h

```diff
@@ -142,6 +142,16 @@ class filter_iteratort
     return tmp;
   }
 
+  value_type &operator*()
```

Comment:
:question: The commit message + change seem unrelated to me, perhaps you could expand in the commit message why these are required.